rispy/parser.py

```diff
@@ -335,14 +350,27 @@ def load(
                                  Consistent with the python standard library,
                                  if `None` is supplied, the default system
                                  encoding is used.
+        newline(str, optional): File line separator.
         implementation (RisImplementation): RIS implementation; base by
                                             default.
 
     Returns:
         list: Returns list of RIS entries.
     """
-    text = file.read_text(encoding=encoding) if isinstance(file, Path) else file.read()
-    return loads(text, implementation=implementation, **kw)
+    if implementation is None:
+        parser = RisParser
+    else:
+        parser = implementation
+
+    if hasattr(file, "readline"):
+        return parser(newline=newline, **kw).parse_lines(file)
+    elif hasattr(file, "open"):
+        with file.open(mode="r", newline=newline, encoding=encoding) as f:
+            return parser(**kw).parse_lines(f)
+    elif hasattr(file, "read"):
+        return loads(file.read(), implementation=implementation, newline=newline, **kw)
+    else:
+        raise ValueError("File must be a file-like object or a Path object")
```

Comment:
I don't completely understand what functionality is added here.

Reply:
This block stems from us allowing multiple types of inputs (file-like objects and paths. To make it more flexible, I decided to check for different methods instead of the type.

* if a user supplies an open file that has a readline method, we can iterate over it
* If a user supplies a path, we need to open it first
* There are some file-like objects that don’t have a readline method. As a backup, we can read the whole thing into a string and process it all at once


I’m open to other ways of doing this if you have any ideas!